src/fsharp/FSharp.Core/fslib-extra-pervasives.fs

```diff
@@ -70,6 +81,78 @@ module ExtraTopLevelOperators =
                     let key = RuntimeHelpers.StructBox(k)
                     if d.ContainsKey(key) then (r <- d.[key]; true) else false
                 member s.Remove(k : 'Key) = (raise (NotSupportedException(SR.GetString(SR.thisValueCannotBeMutated))) : bool) 
+#if FX_NO_READONLY_COLLECTIONS
```

Comment:
Please use feature-based defines (FX_NO_READONLY_COLLECTIONS) rather than platform defines (FX_ATLEAST_45, FSHARP_CORE_PORTABLE).  Granularity of "features" is up to you - merge related ones where possible, split where necessary,

thanks
Don